pkg/lib/queueinformer/queueinformer.go

```diff
@@ -105,10 +111,26 @@ func NewInformer(queue workqueue.RateLimitingInterface, informer cache.SharedInd
 		name:            name,
 		MetricsProvider: metrics,
 	}
-	if funcs == nil {
-		queueInformer.resourceEventHandlerFuncs = queueInformer.defaultResourceEventHandlerFuncs()
-	} else {
-		queueInformer.resourceEventHandlerFuncs = funcs
+	queueInformer.resourceEventHandlerFuncs = queueInformer.defaultResourceEventHandlerFuncs()
+	if funcs != nil {
+		if funcs.AddFunc != nil {
```

Comment:
Do we always want to run the default logic? I would think we'd want to overwrite it if you pass one in.

Reply:
I agree.